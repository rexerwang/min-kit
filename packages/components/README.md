# @min-kit/components

[![](https://img.shields.io/codecov/c/github/rexerwang/min-kit?flag=components&style=for-the-badge)](https://codecov.io/gh/rexerwang/min-kit/flags)
[![](https://img.shields.io/npm/types/%40min-kit/components?style=for-the-badge)](https://github.com/rexerwang/min-kit/tree/main/packages/components)
[![](https://img.shields.io/npm/v/%40min-kit/components?style=for-the-badge)](https://npm.im/@min-kit/components)
[![](https://img.shields.io/badge/React-Tarojs-007ACC?style=for-the-badge&logo=react&logoColor=61DAFB&labelColor=20232A)](https://github.com/NervJS/taro)

React components for miniapp

## Usage

```tsx
import { MinIcon, MinImage, MinNavigationBar, Modal, type ReactProps } from '@min-kit/components'
import { go } from '@min-kit/extends'
import { Button, View } from '@tarojs/components'

// pages/index/index.tsx
export default function Index(props: ReactProps) {
  const openModal = () => {
    Modal.confirm({
      title: 'Confirm',
      subtitle: 'the subtitle\nsupport multiple lines.',
      okText: 'ok',
      cancelText: 'cancel',
    })
  }

  return (
    <View className='relative box-border'>
      <MinNavigationBar
        title={
          <View className='flex-center-y'>
            <MinIcon name='back' size={20} onClick={() => go.back()} />
            <View className='ml-3 text-sm'>@min-kit/components</View>
          </View>
        }></MinNavigationBar>

      <View className='flex flex-col items-center'>
        <MinImage
          className='w-[750px] h-[750px]'
          src='https://source.unsplash.com/random/400x400?nature'
          preview={[
            'https://source.unsplash.com/random/400x400?nature',
            'https://source.unsplash.com/random/400x400?people',
            'https://source.unsplash.com/random/400x400?foods',
          ]}
        />
        <Button className='mt-4' type='primary' size='mini' plain onClick={openModal}>
          open modal
        </Button>
      </View>
    </View>
  )
}

definePageConfig({
  navigationStyle: 'custom',
})
```

### Import styles

```js
// Import overall styles
import '@min-kit/components/dist/styles/index.css'

// Import individual component styles
import '@min-kit/components/dist/styles/min-debugger.css'
import '@min-kit/components/dist/styles/min-icon.css'
import '@min-kit/components/dist/styles/min-image.css'
import '@min-kit/components/dist/styles/min-navigation-bar.css'
import '@min-kit/components/dist/styles/modal.css'

// Import SCSS
import '@min-kit/components/styles/index.scss'
import '@min-kit/components/styles/min-debugger.scss'
import '@min-kit/components/styles/min-icon.scss'
import '@min-kit/components/styles/min-image.scss'
import '@min-kit/components/styles/min-navigation-bar.scss'
import '@min-kit/components/styles/modal.scss'
```

More usage examples ðŸ‘‰ https://github.com/rexerwang/min-kit/tree/main/packages/example
