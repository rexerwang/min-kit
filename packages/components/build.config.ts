import { exec } from 'node:child_process'
import { resolve } from 'node:path'
import { promisify } from 'node:util'

import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  clean: !process.argv.includes('--no-clean'),
  entries: [
    'src/index',
    {
      builder: 'mkdist',
      input: 'styles',
      outDir: 'dist/styles',
    },
  ],
  declaration: true,
  rollup: {
    emitCJS: true,
    esbuild: {
      jsx: 'automatic',
    },
  },
  hooks: {
    async 'mkdist:done'() {
      const input = resolve(__dirname, 'dist/styles')
      await promisify(exec)(`pnpm postcss ${input} -d ${input}`)
    },
  },
})
