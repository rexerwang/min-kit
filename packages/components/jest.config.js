module.exports = require('@min-kit/jest/config')({
  collectCoverageFrom: ['!src/debugger/{hooks,service,store}/**'],
})
