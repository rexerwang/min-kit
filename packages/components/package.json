{
  "name": "@min-kit/components",
  "version": "0.5.1",
  "description": "React components for miniapp",
  "author": "rexerwang",
  "license": "MIT",
  "repository": "rexerwang/min-kit",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.cjs"
    },
    "./dist/styles/*.css": {
      "import": "./dist/styles/*.css",
      "require": "./dist/styles/*.css"
    }
  },
  "main": "./dist/index.cjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "files": [
    "dist",
    "styles"
  ],
  "keywords": [
    "min-kit",
    "components",
    "react",
    "taro",
    "miniapp",
    "miniprogram"
  ],
  "scripts": {
    "build": "unbuild",
    "test": "jest"
  },
  "dependencies": {
    "@min-kit/extends": "workspace:*",
    "@min-kit/hooks": "workspace:*",
    "@min-kit/shared": "workspace:*",
    "@min-kit/store": "workspace:*",
    "clsx": "^2.1.0"
  },
  "peerDependencies": {
    "@tarojs/components": ">=3.6",
    "@tarojs/plugin-platform-alipay": ">=3.6",
    "@tarojs/plugin-platform-h5": ">=3.6",
    "@tarojs/plugin-platform-jd": ">=3.6",
    "@tarojs/plugin-platform-qq": ">=3.6",
    "@tarojs/plugin-platform-swan": ">=3.6",
    "@tarojs/plugin-platform-tt": ">=3.6",
    "@tarojs/plugin-platform-weapp": ">=3.6",
    "@tarojs/react": ">=3.6",
    "@tarojs/taro": ">=3.6",
    "react": ">=17",
    "react-dom": ">=17"
  },
  "devDependencies": {
    "@min-kit/helper": "workspace:*",
    "@min-kit/jest": "workspace:^"
  }
}
