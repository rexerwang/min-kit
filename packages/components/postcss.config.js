module.exports = {
  plugins: {
    tailwindcss: {},
    'postcss-rem-to-responsive-pixel': {
      rootValue: 32,
      propList: ['*'],
      transformUnit: 'rpx',
    },
  },
}
