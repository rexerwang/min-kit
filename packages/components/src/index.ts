export * from './debug'
export * from './gesture'
export * from './icon'
export * from './img'
export * from './modal'
export * from './modal/drawer'
export * from './navigation-bar'
export * from './portal'
export * from './share'
export * from './text'
export type { ReactProps } from './types'
