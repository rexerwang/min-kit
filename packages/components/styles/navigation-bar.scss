.mini-navigation-bar {
  position: relative;
  box-sizing: border-box;
  width: 100vw;

  &-body {
    margin-right: 16px;
    margin-left: 16px;
  }
}
