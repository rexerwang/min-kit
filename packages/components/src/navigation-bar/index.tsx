import { SystemInfo } from '@miniapp/extends'
import { px } from '@miniapp/shared'
import { View } from '@tarojs/components'
import clsx from 'clsx'

import { ReactProps } from '../types'

const paddingTop = SystemInfo.menuButtonRect.top ?? SystemInfo.statusBarHeight,
  marginBottom = 10,
  width = SystemInfo.menuButtonRect.left,
  height = SystemInfo.menuButtonRect.height

export interface NavigationBarProps extends ReactProps {
  title?: React.ReactNode
}

export function NavigationBar({ className, children, title }: NavigationBarProps) {
  return (
    <View className={clsx('mini-navigation-bar ', className)} style={px.of({ paddingTop })}>
      <View className='mini-navigation-bar-body' style={px.of({ width, height, marginBottom })}>
        {title}
      </View>
      {children}
    </View>
  )
}

NavigationBar.displayName = 'MiniNavigationBar'
NavigationBar.rect = {
  height,
  width,
  top: paddingTop,
  bottom: height + paddingTop + marginBottom,
}
