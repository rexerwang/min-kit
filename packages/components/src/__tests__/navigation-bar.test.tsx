import { render } from '@min-kit/jest'
import { View } from '@tarojs/components'

import { MinNavigationBar } from '../navigation-bar'

describe('<MinNavigationBar />', () => {
  it('should render <MinNavigationBar /> toMatchSnapshot', async () => {
    const { asFragment } = render(
      <MinNavigationBar className='nav' title='min-kit'>
        <View>MinNavigationBar</View>
      </MinNavigationBar>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('MinNavigationBar.rect', () => {
    expect(MinNavigationBar.rect).toEqual(
      expect.objectContaining({
        height: expect.any(Number),
        width: expect.any(Number),
        top: expect.any(Number),
        bottom: expect.any(Number),
      }),
    )
  })
})
