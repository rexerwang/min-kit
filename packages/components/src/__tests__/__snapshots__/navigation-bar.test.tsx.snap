// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MinNavigationBar /> should render <MinNavigationBar /> toMatchSnapshot 1`] = `
<DocumentFragment>
  <taro-view-core
    class="min-navigation-bar nav"
    style="padding-top: 48px;"
  >
    <!---->
    <taro-view-core
      class="min-navigation-bar-body"
      style="width: 281px; height: 32px; margin-bottom: 10px;"
    >
      <!---->
      min-kit
    </taro-view-core>
    <taro-view-core
      class=""
    >
      <!---->
      MinNavigationBar
    </taro-view-core>
  </taro-view-core>
</DocumentFragment>
`;
